modules/ppcp-wc-gateway/src/Helper/RefundFeesUpdater.php

```diff
@@ -0,0 +1,193 @@
+		$currency_codes     = array();
+		$refunds_ids        = array();
+
+		foreach ( $purchase_units as $purchase_unit ) {
+			$payments = $purchase_unit->payments();
+
+			if ( ! $payments ) {
+				continue;
+			}
+
+			$refunds = $payments->refunds();
+
+			foreach ( $refunds as $refund ) {
+				$breakdown     = $refund->seller_payable_breakdown();
+				$refunds_ids[] = $refund->id();
+
+				if ( ! $breakdown ) {
+					continue;
+				}
+
+				$gross_amount = $breakdown->gross_amount();
+				if ( $gross_amount ) {
+					$gross_amount_total += $gross_amount->value();
+					$currency_codes[]    = $gross_amount->currency_code();
+				}
+
+				$paypal_fee = $breakdown->paypal_fee();
+				if ( $paypal_fee ) {
+					$fee_total       += $paypal_fee->value();
+					$currency_codes[] = $paypal_fee->currency_code();
+				}
+
+				$net_amount = $breakdown->net_amount();
+				if ( $net_amount ) {
+					$net_amount_total += $net_amount->value();
+					$currency_codes[]  = $net_amount->currency_code();
+				}
+			}
+		}
+
+		$currency_codes = array_unique( $currency_codes );
+
+		if ( count( $currency_codes ) > 1 ) {
+			// There are multiple different currencies codes in the refunds.
+
+			$this->logger->warning(
+				sprintf(
+					'Updating order paypal refund fees. Multiple currencies detected. [wc_order: %s, paypal_order: %s, currencies: %s]',
+					$wc_order->get_id(),
+					$paypal_order_id,
+					implode( ',', $currency_codes )
+				)
+			);
+
+			$wc_order->update_meta_data( PayPalGateway::REFUND_FEES_META_KEY, array() );
+			return;
+		}
+
+		$currency_code = current( $currency_codes ) ?: '';
+
+		$meta_data = array(
+			'gross_amount' => ( new Money( $gross_amount_total, $currency_code ) )->to_array(),
+			'paypal_fee'   => ( new Money( $fee_total, $currency_code ) )->to_array(),
+			'net_amount'   => ( new Money( $net_amount_total, $currency_code ) )->to_array(),
+		);
+
+		$wc_order->update_meta_data( PayPalGateway::REFUND_FEES_META_KEY, $meta_data );
+		$wc_order->save();
+
+		$order_notes = $this->get_order_notes( $wc_order );
+
+		foreach ( $refunds_ids as $refund_id ) {
+			$has_note = false;
+			foreach ( $order_notes as $order_note ) {
+				if ( strpos( $order_note->comment_content, $refund_id ) !== false ) {
+					$has_note = true;
+				}
+			}
+			if ( ! $has_note ) {
+				$wc_order->add_order_note( sprintf( 'PayPal refund ID: %s', $refund_id ) );
+			}
+		}
+
+		$this->logger->debug(
+			sprintf( 'Updated order paypal refund fees. [wc_order: %s, paypal_order: %s]', $wc_order->get_id(), $paypal_order_id )
+		);
+	}
+
+	/**
+	 * Returns all order notes
+	 * Based on WC_Order::get_customer_order_notes
+	 *
+	 * @param WC_Order $wc_order The WooCommerce order.
+	 * @return WP_Comment[]
+	 */
+	private function get_order_notes( WC_Order $wc_order ): array {
+		$notes = array();
+		$args  = array(
+			'post_id' => $wc_order->get_id(),
+		);
+
+		remove_filter( 'comments_clauses', array( 'WC_Comments', 'exclude_order_comments' ) );
```

Comment:
Would be great to add a comment why this is needed.

Reply:
The function like stated on the comments is based on https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/class-wc-order.php#L1962-L1984, WooCommerce doesn't have a comment in their code. But I'll add one.